ocaml/xapi/xapi_pool_update.ml

```diff
@@ -340,18 +340,31 @@ let introduce ~__context ~vdi =
 let pool_apply ~__context ~self =
   let pool_update_name = Db.Pool_update.get_name_label ~__context ~self in
   debug "pool_update.pool_apply %s" pool_update_name;
-  ()
+
+  let applied_hosts = Db.Pool_update.get_hosts ~__context ~self in
```

Comment:
I think this is a good opportunity to use a pipeline here:

    Db.Pool_update.get_hosts ~__context ~self in
    |> List.set_difference (Db.Host.get_all ~__context)
    |> List.iter ...

This safes you coming up with names for intermediate results.